lsp/document.go

```diff
@@ -0,0 +1,54 @@
+package lsp
+
+import (
+	"strings"
+
+	protocol "github.com/tliron/glsp/protocol_3_16"
+)
+
+type document struct {
+	uri     protocol.DocumentUri
+	Content string
+	lines   []string
+}
+
+type documentStore struct {
```

Comment:
I assume this is caching files that the LSP is interacting with? If so why do we need the `os.ReadFile()` above?

But more broadly, is this a good idea? Will this end up keeping a copy of the entire Git repo in RAM? Or is it just open files in the editor?

Reply:
Yeah, the store is set up to `set` and `delete` documents as they are opened and closed (or changed for updates). So it should only have the files that are actually open in RAM.

My goal was to not have to read in the document for all the interactions like `hover` etc. So we already have them preloaded.